network-api/networkapi/wagtailpages/tests/buyersguide/test_consumer_creepometer_page.py

```diff
@@ -0,0 +1,64 @@
+from networkapi.wagtailpages import models as pagemodels
+from networkapi.wagtailpages.factory import buyersguide as buyersguide_factories
+from networkapi.wagtailpages.tests import base as test_base
+
+
+class ConsumerCreepometerPageFactoryTests(test_base.WagtailpagesTestCase):
+    def test_factory(self):
+        buyersguide_factories.ConsumerCreepometerPageFactory()
+
+
+class ConsumerCreepometerPageTests(test_base.WagtailpagesTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.buyersguide_homepage = buyersguide_factories.BuyersGuidePageFactory(
+            parent=cls.homepage,
+        )
+        cls.content_index = buyersguide_factories.BuyersGuideEditorialContentIndexPageFactory(
+            parent=cls.buyersguide_homepage,
+        )
+
+    def test_parents(self):
+        self.assertAllowedParentPageTypes(
+            child_model=pagemodels.BuyersGuideArticlePage,
+            parent_models={pagemodels.BuyersGuideEditorialContentIndexPage},
+        )
+
+    def test_children(self):
+        self.assertAllowedSubpageTypes(
+            parent_model=pagemodels.ConsumerCreepometerPage,
+            child_models={},
+        )
+
+    def test_year_choices(self):
+        page = buyersguide_factories.ConsumerCreepometerPageFactory(
+            parent=self.content_index, title="Test Consumer Creepometer Page", year="2023"
+        )
+        choices = page._meta.get_field("year").choices
+        self.assertEqual(choices, (("2023", "2023"),))
```

Comment:
Here, I would like to make this fail on the factory level when we add more year choices to make sure we remember to add more choices there:

```python
page = buyersguide_factories.ConsumerCreepometerPageFactory(parent-self.content_index)  # it will use a random year
choices = page._meta.get_field("year").choices
self.assertEqual(choices, (("2023", "2023"),))
self.assertIn((page.year, page.year), choices)
```

Reply:
Great suggestions!